### PR TITLE
fix(component): nested modal view-transition support

### PR DESCRIPTION
**Pull request by CatsJuice**
<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/LakojjjzZNf6ogjOVwKE/b8033923-8f08-4c72-96b6-d790fc0ff703.mp4">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/LakojjjzZNf6ogjOVwKE/b8033923-8f08-4c72-96b6-d790fc0ff703.mp4">CleanShot 2024-09-03 at 18.36.17.mp4</video>

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#8073** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8073?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8066** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8066?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by forehalo — COMMENTED** [review]
[bow](https://app.graphite.dev//settings/meme-library?org=toeverything)

**Review by forehalo — APPROVED** [review]

File: packages/frontend/component/src/ui/modal/modal.stories.tsx [hunk 1]
```diff
@@ -139,3 +139,37 @@ export const Animations = () => {
     </div>
   );
 };
+
+export const Nested = () => {
+  const [openRoot, setOpenRoot] = useState(false);
+  const [openNested, setOpenNested] = useState(false);
```
Comment (forehalo) [inline packages/frontend/component/src/ui/modal/modal.stories.tsx:145]: we definitely need `[visible, show, close] = useToggle(false)`

Comment (CatsJuice) [reply packages/frontend/component/src/ui/modal/modal.stories.tsx:145]: If so, `toggle()` is also necessary 🥲

**Review by pengx17 — APPROVED** [review]

**Pull request merged**